### PR TITLE
Code style upgrade

### PR DESCRIPTION
**Pull request by websemantics**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks backward-compatibility?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any
| Updated README/docs?   | ❌
| Added CHANGELOG entry?   | ✔️

I find this project very useful especially how clean the code is and free of dependencies.
I made few changes to the core code in a project I am working on and I wanted to share,

Here's a summary,

- Removed `Class.js` in favour of Javascript class declaration.
- Ability to add style rules for editors (i.e. `StarratingEditor`)
- Modified `build/CSSToJson.js` to generate Javascript modules instead of Json files. This is helpful when importing `JSONEditor` directly as Javascript module without a build step.
- Unified how styles are loaded for themes and editors,
	- Style rules are stored in class static property ‘rules’
	- Rules are loaded as ESM Module,
    		- Generated @ build stage `stylesheet-filename.css.js` or
    		- Written manually.
- Cleaned `src/defaults.js` and moved resolvers to a separate module, `src/resolvers.js`
- Changed how default `themes`, `iconlibs`, `editors` and `templates` are imported to JSONEditor

The code builds without errors and works from `dist` or `src` directly (imported as a module)

Let me know what you think,

### CODE REVIEW

**Comment by websemantics** [comment]
Added ability to attache editors and themes style rules to the shadowRoot if the editor is inside a Web Component

**Comment by schmunk42** [comment]
Can you have a look at the tests, they fail with

```
ERROR in ./tests/unit/class.spec.js
Module not found: Error: Can't resolve '../../src/class' in '/home/travis/build/json-editor/json-editor/tests/unit'
 @ ./tests/unit/class.spec.js 1:0-40 5:11-16
```

**Comment by websemantics** [comment]
That's fixed plus few other related failed tests.
All tests pass on my end.

File: config/webpack.common.js [hunk 1]
```diff
@@ -21,14 +21,14 @@ module.exports = {
   },
   module: {
     rules: [
+      // {
```
Comment (schmunk42) [inline config/webpack.common.js]: Why is this commented?

File: config/webpack.prod.js [hunk 2]
```diff
@@ -9,12 +9,12 @@ const ENV = (process.env.NODE_ENV = process.env.ENV = 'production')
 function createConfig (target) {
   filenameInsert = target === 'var' ? '.' : '.' + target + '.'
   commonConfig.module.rules = [
-    {
-      enforce: 'pre',
-      test: /\.js$/,
-      exclude: /node_modules/,
-      loader: `eslint-loader`
-    },
+    // {
```
Comment (schmunk42) [inline config/webpack.prod.js]: Same here, why commented?

File: src/editors/selectize.js [hunk 3]
```diff
@@ -1,106 +1,113 @@
-import { SelectEditor } from './select'
-import { $extend } from '../utilities'
-export var SelectizeEditor = SelectEditor.extend({
+import  { SelectEditor } from  './select.js'
```
Comment (schmunk42) [inline src/editors/selectize.js]: File modes are changed to 755, they should be 644

File: config/webpack.common.js (hunk 1 above)
Comment (websemantics) [reply config/webpack.common.js]: I wasn't sure how to proceed with eslint as there were large changes to the syntax. I can uncomment and try resolve the warnings?

File: src/editors/selectize.js (hunk 3 above)
Comment (websemantics) [reply src/editors/selectize.js]: Not sure why but that's an easy fix.

**Comment by germanbisurgi** [comment]
Hi websemantics and thanks very much for sharing!

I've been able to give a peek at all the changes you've been proposing. Everything looks good and the tests pass.

Unfortunately, I'll never be able to do a meaningful code review because there is too much goodness in you PR :)

I'm wondering if we could split all the extra improvements and make gradual changes in order to be able to catch bug while appearing.

File: config/webpack.common.js (hunk 1 above)
Comment (schmunk42) [reply config/webpack.common.js]: Yes please, we need to run eslint.

**Comment by websemantics** [comment]
Hi germanbisurgi, and thanks for your remarks.

I agree, there are too many changes and I was testing excessively with the app I am working on to ensure it didn't break. But you're right, one can't be too careful.

Though, I am not sure how to break this meaningfully since and the main reason for the huge amount of changes was the removal of Class.js which meant all files were changed to use Javascript class declaration/inheritance.

I was tempted to remove `const self=this` as that no longer is required with the use of arrow function but I didn't because that would've been a major source of bugs if function scope was intentional.

Let me know if you have suggestions and I'll be glad to do it.

**Comment by schmunk42** [comment]
@germanbisurgi Didn't we do a lot of fixes already in the source code? I wonder why things like "semicolon" are still popping up.

Or @websemantics was this introduced in this PR?

**Comment by websemantics** [comment]
Hi schmunk42, not sure. 

I don't use semicolon but also, my eslint settings are different. 

To double check,  I just cloned a fresh json-editor/json-editor and ran the build script and I am getting eslint errors. I think that was why I initially concluded to comment them. 

Not sure if that was helpful but let me know if I needed to do anything on my end.

**Comment by schmunk42** [comment]
@germanbisurgi What's the current eslint status, when running tests/builds on Travis?

**Comment by germanbisurgi** [comment]
It extends "standard" and has "no-undef":"off" rule for some reason

**Pull request merged**